oauthenticator/github.py

```diff
@@ -113,6 +114,19 @@ def _github_client_secret_changed(self, name, old, new):
         config=True, help="Automatically allow members of selected organizations"
     )
 
+    populate_teams_in_auth_state = Bool(
+        False,
+        help="""
+        If auth_state is enabled, populate it with the list of teams the current user is a member of.
+
+        'teams' will be a key in auth_state that has the same structure listed
+        in https://docs.github.com/en/rest/reference/teams#list-teams-for-the-authenticated-user.
+
+        Requires `read:org` to be set in scopes.
```

Comment:
I wanted to clarify that we populate authentication state no matter if `enable_auth_state` is configured. This state, while not persisted without `enable_auth_state`, can still be relevant to use by a `post_auth_hook`, so I don't think we should add a condition about only doing this if `enable_auth_state` is set either.

Yet, since the authentication state is perhaps mostly relevant for other purposes outside of the `post_auth_hook`, it would be good to link out to authentication state documentation where `enable_auth_state` is described in more detail.

```suggestion
        Populates the authentication state dictionary `auth_state` with a key
        `teams` assigned the list of teams the current user is a member of at
        the time of authentication. The teams have the same structure listed in
        https://docs.github.com/en/rest/reference/teams#list-teams-for-the-authenticated-user.

        Requires `read:org` to be set in `scopes`.
        
        Note that authentication state is only be available to a
        `post_auth_hook` before being discarded unless configured to be
        persisted via `enable_auth_state`. For more information, see
        https://jupyterhub.readthedocs.io/en/stable/reference/authenticators.html#authentication-state.
```